aws-gitlab-main.yml

```diff
@@ -374,12 +438,54 @@ Resources:
       Engine: redis
       EngineVersion: '5.0.6' #newest
       Port: 6379 
-      CacheNodeType: !Ref CacheNodeTypeParameter
+      CacheNodeType: !Ref GitLabCacheNodeTypeParameter
       CacheSubnetGroupName: !Ref GitLabRedisSubnetGroup
       PreferredAvailabilityZones: 
         - !GetAtt GitLabPrivateSubnetA.AvailabilityZone 
       AZMode: single-az #does multi-az == multi-AZ auto-failover?
       #had to put only one Az because had to choose 1 NumCacheNodes when using Redis  
       VpcSecurityGroupIds: 
         - !GetAtt GitLabRedisSecurityGroup.GroupId
-  
\ No newline at end of file
+ 
+  #BASTION HOSTS
+  GitLabBastionHostA:
+    Type: AWS::EC2::Instance
+    Properties: 
+      AvailabilityZone: !Ref BastionHostAAvailabilityZoneParameter 
+      
+      ImageId: !Ref   GitLabBastionHostAAMIParameter #ubuntu server 18.04 LTS (HVM) AMI 
+      InstanceType: !Ref GitLabBastionHostAInstanceTypeParameter
+      SubnetId: !Ref GitLabPublicSubnetA
+      #NetworkInterfaces:
+      #   - AssociatePublicIpAddress: "true" #enable auto-assign public IP 
+      # BlockDeviceMappings:  #/dev/sdh or xvdh? 
```

Comment:
depends on OS being used. Unless instructions specify custom storage space. You can avoid this and it will use the default of the AMI